### PR TITLE
fix(providers): add OTEL for messaging kafka provider

### PR DESCRIPTION
**Pull request by vados-cosmonic**
## Feature or Problem

This commit ensures OTEL is working for the messaging-kafka provider.

image

### CODE REVIEW

File: examples/rust/components/echo-messaging/wadm.yaml [hunk 1]
```diff
@@ -31,4 +31,5 @@ spec:
     - name: nats
       type: capability
       properties:
-        image: ghcr.io/wasmcloud/messaging-nats:canary
+        image: file://../../../../messaging-kafka-provider.par
```
Comment (rvolosatovs) [inline examples/rust/components/echo-messaging/wadm.yaml:34]: 🤔 what's the reason for this change?

File: crates/provider-messaging-kafka/src/lib.rs [hunk 2]
```diff
@@ -298,20 +323,35 @@ impl exports::wasmcloud::messaging::consumer::Handler<Option<Context>> for Kafka
...
+                .map(|(k, v)| (k.to_string(), v.to_string()))
+                .collect::<Vec<(String, String)>>(),
+
+            _ => TraceContextInjector::default_with_span()
+                .iter()
+                .map(|(k, v)| (k.to_string(), v.to_string()))
+                .collect(),
+        };
+        wasmcloud_tracing::context::attach_span_context(&trace_ctx);
+        debug!(?subject, ?body, ?timeout_ms, "performing request");
```
Comment (rvolosatovs) [inline crates/provider-messaging-kafka/src/lib.rs:353]: I think this statement is confusing, since nothing is actually being done

Comment (vados-cosmonic) [reply crates/provider-messaging-kafka/src/lib.rs:353]: Sure -- will remove!

- [ ] remove performing request debug

**Review by rvolosatovs — APPROVED** [review]

**Review by ricochet — APPROVED** [review]
